packages/obojobo-document-engine/ObojoboDraft/Sections/Assessment/components/rubric/editor.js

```diff
@@ -0,0 +1,405 @@
+										)
+										break
+									case 1:
+										block = Block.create(
+											ParameterNode.helpers.oboToSlate({
+												name: 'passedResult',
+												value: 100 + '',
+												display: 'Passed Result'
+											})
+										)
+										break
+									case 2:
+										block = Block.create(
+											ParameterNode.helpers.oboToSlate({
+												name: 'failedResult',
+												value: 0 + '',
+												display: 'Failed Result'
+											})
+										)
+										break
+									case 3:
+										block = Block.create(
+											ParameterNode.helpers.oboToSlate({
+												name: 'unableToPassResult',
+												value: '',
+												display: 'Unable to Pass Result'
+											})
+										)
+										break
+								}
+								return change.insertNodeByKey(node.key, index, block)
+							})
+						}
+					}
+				}
+			},
+			'ObojoboDraft.Sections.Assessment.Rubric.ModList': {
+				nodes: [{ match: [{ type: MOD_NODE }], min: 1, max: 20 }],
+				normalize: (change, error) => {
+					const { node, child, index } = error
+					switch (error.code) {
+						case CHILD_REQUIRED: {
+							const block = Block.create({
+								type: MOD_NODE
+							})
+							return change.insertNodeByKey(node.key, index, block)
+						}
+						case CHILD_TYPE_INVALID: {
+							return change.wrapBlockByKey(child.key, {
+								type: MOD_NODE
+							})
+						}
+					}
+				}
+			},
+			'ObojoboDraft.Sections.Assessment.Rubric.Mod': {
+				nodes: [{ match: [{ type: 'Parameter' }], min: 2, max: 2 }],
+				normalize: (change, error) => {
+					const { node, child, index } = error
+					switch (error.code) {
+						case CHILD_REQUIRED: {
+							if (index === 0) {
+								const block = Block.create(
+									ParameterNode.helpers.oboToSlate({
+										name: 'attemptCondition',
+										value: '[1,$last_attempt]',
+										display: 'Attempt Condition'
+									})
+								)
+								return change.insertNodeByKey(node.key, index, block)
+							}
+							const block = Block.create(
+								ParameterNode.helpers.oboToSlate({
+									name: 'reward',
+									value: 0 + '',
+									display: 'Reward'
+								})
+							)
+							return change.insertNodeByKey(node.key, index, block)
+						}
+						case CHILD_TYPE_INVALID: {
+							return change.withoutNormalization(c => {
+								c.removeNodeByKey(child.key)
+								if (index === 0) {
+									const block = Block.create(
+										ParameterNode.helpers.oboToSlate({
+											name: 'attemptCondition',
+											value: '[1,$last_attempt]',
+											display: 'Attempt Condition'
+										})
+									)
+									return c.insertNodeByKey(node.key, index, block)
+								}
+								const block = Block.create(
+									ParameterNode.helpers.oboToSlate({
+										name: 'reward',
+										value: 0 + '',
+										display: 'Reward'
+									})
+								)
+								return c.insertNodeByKey(node.key, index, block)
+							})
+						}
+					}
```

Comment:
Similar comment, above code has two similar pieces, is it possible to consolidate them?

Reply:
Not currently, because of the need to wrap the second half in change.withoutNormalizing